scripts/generate_object_properties.cr

```diff
@@ -0,0 +1,894 @@
+#! /usr/bin/env crystal
+
+struct Generator
+  def initialize(@file : File, @macro_prefix : String, @method_prefix : String, @var_prefix : String, @doc_prefix : String)
+  end
+
+  def puts
+    @file.puts
+  end
+
+  def puts(message)
+    @file.puts(message)
+  end
+
+  def def_vars
+    <<-TEXT
+          {% if name.is_a?(TypeDeclaration) %}
+            {% var_name = name.var.id %}
+            {% type = name.type %}
+            {% if block %}
+              #{@var_prefix}{{var_name}} : {{type}}? {% if name.value %} = {{name.value}} {% end %}
+            {% else %}
+              #{@var_prefix}{{name}}
+            {% end %}
+          {% elsif name.is_a?(Assign) %}
+            {% var_name = name.target %}
+            {% type = nil %}
+            #{@var_prefix}{{name}}
+          {% else %}
+            {% var_name = name.id %}
+            {% type = nil %}
+          {% end %}
+
+    TEXT
+  end
+
+  def def_getter(suffix = "")
+    if @macro_prefix == "class_"
+      def_class_getter suffix
+    else
+      <<-TEXT
+            def {{var_name}}#{suffix} {% if type %} : {{type}} {% end %}
+              {% if block %}
+                if (%value = @{{var_name}}).nil?
+                  @{{var_name}} = {{yield}}
+                else
+                  %value
+                end
+              {% else %}
+                @{{var_name}}
+              {% end %}
+            end
+
+      TEXT
+    end
+  end
+
+  def def_class_getter(suffix = "")
+    <<-TEXT
+            {% if block %}
+              {% if ::Crystal::Once::FlagT == Bool %}
+                @@__{{var_name}}_flag = false
+              {% else %}
+                @@__{{var_name}}_flag = ::Crystal::Once::State::Uninitialized
+              {% end %}
```

Comment:
I think this answers my question in https://github.com/crystal-lang/crystal/pull/15370#issuecomment-2615923122

I could use the `FlagT` alias directly but it needs an initial value, so it would mean another `FLAG_DEFAULT` constant. Yet, having a single implementation and a simple `Bool` would be simpler.